### PR TITLE
feat(277): upgrade dependency management go modules

### PR DESCRIPTION
**Pull request by clD11**
This PR migrates Kelp from using glide to go modules for our dependency management

* Glide has been removed
* go.mod added
* Readme updated
* CircleCI updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,25 +124,17 @@ _Note for Windows Users: You should use a [Bash Shell][bash] to follow the steps
...
-    * Change to `kelp` directory for all following commands: `cd $GOPATH/src/github.com/stellar/kelp`
-5. Install the dependencies:
-    * `glide install`
-6. Install the [astilectron-bundler][astilectron-bundler] binary into `$GOBIN`
-    * `go get -u github.com/asticode/go-astilectron-bundler/...`
+1. [Download][golang-download] and [setup][golang-setup] Golang _v1.13 or later_
+   1. Set environment variable `export GOPROXY=https://proxy.golang.org`
+2. Install [Yarn][yarn-install] and [NodeJs][nodejs-install] to build the Kelp GUI
+3. Clone the kelp repository `git clone git@github.com:stellar/kelp.git`
+4. Install the [astilectron-bundler][astilectron-bundler] binary into `$GOPATH/bin`
```
Comment (nikhilsaraf) [inline README.md:131]: We do not define where to set GOPATH, do we need to or can we remove it?

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,20 +5,17 @@ commands:
...
-          command: sudo apt-get update -y && sudo apt-get install -y golang-glide
-      - run:
-          name: Install dependencies using glide
-          command: glide install
-      - run:
-          name: Fetch astilectron-bundler
-          command: go get -u github.com/asticode/go-astilectron-bundler/...
+          name: Download Go Modules
+          command: |
+            export GOPROXY=https://goproxy.io,https://proxy.golang.org,https://goproxy.cn
```
Comment (nikhilsaraf) [inline .circleci/config.yml:13]: We should test only with proxy.Golang.org

File: .circleci/config.yml [hunk 3]
```diff
@@ -5,20 +5,17 @@ commands:
...
-          name: Install glide
-          command: sudo apt-get update -y && sudo apt-get install -y golang-glide
-      - run:
-          name: Install dependencies using glide
-          command: glide install
-      - run:
-          name: Fetch astilectron-bundler
-          command: go get -u github.com/asticode/go-astilectron-bundler/...
+          name: Download Go Modules
+          command: |
```
Comment (nikhilsaraf) [inline .circleci/config.yml:12]: Why start out these commands with |
any advantage?

File: scripts/build.sh [hunk 4]
```diff
@@ -120,7 +122,7 @@ function download_file() {
     echo "... downloaded $NAMESPACE to $DESTINATION"
 }
 
-if [[ $(basename $("pwd")) != "kelp" ]]
+if [[ $(basename $("pwd")) != "kelp" && $("pwd") != *"circleci/project"* ]];
```
Comment (nikhilsaraf) [inline scripts/build.sh:125]: Why add the circleci exception now when it wasn’t needed earlier?

File: plugins/ccxtExchange_test.go [hunk 5]
```diff
@@ -556,24 +557,17 @@ func TestCancelOrder_Ccxt(t *testing.T) {
 }
 
 func TestGetOrderConstraints_Ccxt_Precision(t *testing.T) {
-	// coinbasepro gives incorrect precision values so we do not test it here
```
Comment (nikhilsaraf) [inline plugins/ccxtExchange_test.go:559]: Why modify this file?
(Wondering how it’s related to the change)

File: go.mod [hunk 6]
```diff
@@ -0,0 +1,68 @@
+module github.com/stellar/kelp
+
+go 1.13
+
+require (
+	cloud.google.com/go v0.84.0 // indirect
+	github.com/Beldur/kraken-go-api-client v0.0.0-20180126083054-8d8ccfe4cc60
```
Comment (nikhilsaraf) [inline go.mod:7]: Seems reasonable — any dependency version updated here or it’s a direct port of the previous versions?

**Review by nikhilsaraf — COMMENTED** [review]
Some questions. Seems good besides that

File: .circleci/config.yml (hunk 2 above)
Comment (clD11) [reply .circleci/config.yml:13]: I tried that yesterday and it failed so had to add the others, I think its actually this https://goproxy.cn one that has that lib but kept the others for completeness.

File: go.mod (hunk 6 above)
Comment (clD11) [reply go.mod:7]: Direct port all the versions should be the same

File: README.md (hunk 1 above)
Comment (clD11) [reply README.md:131]: I think GOPATH is synonymous with wherever your go workspace is as they will still need to  install astilectron-bundler in either where ever they set GOPATH or default $HOME/go

Comment (clD11) [reply README.md:131]: I need to update the proxy value though

File: .circleci/config.yml (hunk 3 above)
Comment (clD11) [reply .circleci/config.yml:12]: Just when executing multiple individual  commands  over multiple lines I think its cleaner than > or using sh

File: scripts/build.sh (hunk 4 above)
Comment (clD11) [reply scripts/build.sh:125]: There was a very specific reason for this I think its the way the modules install and the new directory structure but it was ages ago and I have forgotten the details :laughing:

File: plugins/ccxtExchange_test.go (hunk 5 above)
Comment (clD11) [reply plugins/ccxtExchange_test.go:559]: just to get the build green

File: README.md (hunk 1 above)
Comment (nikhilsaraf) [reply README.md:131]: Many devs are new to Golang and will get confused by the GOPATH reference here. Go modules doesn’t need GOPATH anymore so we can just suggest that they install kelp into ~/kelp and use that as the path references throughout so it is simple for the reader who is new to Golang

File: .circleci/config.yml (hunk 2 above)
Comment (nikhilsaraf) [reply .circleci/config.yml:13]: Ok then let’s include all 3 in the readme too so it is consistent with the CI setup

File: .circleci/config.yml (hunk 3 above)
Comment (nikhilsaraf) [reply .circleci/config.yml:12]: Ok fair

File: scripts/build.sh (hunk 4 above)
Comment (nikhilsaraf) [reply scripts/build.sh:125]: Ok no worries. Let’s leave it in, if you remember the reason then please add that context as a comment next to the code for posterity :)

File: plugins/ccxtExchange_test.go (hunk 5 above)
Comment (nikhilsaraf) [reply plugins/ccxtExchange_test.go:559]: Once we merge this fix we can rebase and this will work.

**Review by nikhilsaraf — APPROVED** [review]
Lgtm, thanks!

**Pull request merged**